### ISSUE TITLE
Maybe something wrong when using multiple entry

### ISSUE THREAD

**Issue by qmhc** [qmhc/vite-plugin-dts#192]
Link to https://github.com/qmhc/vite-plugin-dts/issues/189#issuecomment-1474769466

**Comment by qmhc** [comment]
@derpdead We can tlak about your question in this issue.

Whether a `index.d.ts` file under 'dist' dir? And what's your `tsconfig.json`?

**Comment by derpdead** [comment]
Hello @qmhc , thank you for your time!

`tsconfig.json`

```
{
  "compilerOptions": {
    "target": "esnext",
    "useDefineForClassFields": true,
    "module": "esnext",
    "moduleResolution": "node",
    "strict": true,
    "jsx": "preserve",
    "allowJs": true,
    "sourceMap": true,
    "outDir": "./dist",
    "resolveJsonModule": true,
    "esModuleInterop": true,
    "lib": ["esnext", "dom"],
    "declaration": true,
    "declarationMap": true,
    "emitDeclarationOnly": true,
    "declarationDir": "./dist/types",
    "isolatedModules": true,
    "skipLibCheck": true,
    "types": [
      "vite/client"
    ],
    "paths": {
      "@": ["./src/*"],
      "@Types": ["./src/types/types"],
      "@Stores/*": ["./src/stores/*"],
      "@Components/*": ["./src/components/*"],
      "@PlaygroundComponents/*": ["./src/playground-components/*"],
      "@Hooks/*": ["./src/hooks/*"],
      "@Utils/*": ["./src/utils/*"],
      "@Services/*": ["./src/services/*"],
      "@Defaults/*": ["./src/defaults/*"]
    }
  },
  "include": ["src/**/*.ts", "src/**/*.d.ts", "src/**/*.vue", "src/index.d.ts"],
  "exclude": ["src/playground-components/**", "**/node_modules", "./node_modules", "./node_modules/*"],
  "references": [{ "path": "./tsconfig.node.json" }]
}
```
And the `index.d.ts` is coming from `unplugin-vue-components/vite` defined in `vite.config.ts`

```
      Components({
        dts: 'src/index.d.ts',
        globs: componentsPaths,
        version: 3,
      })
```

As it is not included in build process, I'm moving it manually into `dist` after the pckg is being build. The build command `"build": "vue-tsc && vite build && cp src/index.d.ts dist"` . 

Maybe I don't need to move it into `dist`? Without it I did not have type hinting and `Go to declaration` in projects that I've installed the pckg ;/

**Comment by qmhc** [comment]
Those files under `dist/types` maybe generated by vue-tsc, I see `"declarationDir": "./dist/types"` in your `tsconfig.json`. You can use `vue-tsc --noEmit` for type checking only, or drop it and only use this plugin (by defult the plugin will check type, and you can disable this feature via `skipDiagnostics: false`).

About the dts file which is generated via `unplugin-vue-components`, you can use `afterBuild` option and copy it into `dist` by yourself.

> Whether a index.d.ts file under 'dist' dir?

Sry my vague description. I mean does a file `dist/index.d.ts` exist?

**Comment by derpdead** [comment]
I've removed from `tsconfig` the `emitDeclarationOnly` property and added `--noEmit`.  I see the types generated for each component. Are they by design mirroring file structure in `dist` folder? 

<img width="950" alt="image" src="https://user-images.githubusercontent.com/46050724/226441903-0e0c5785-a124-4549-87ec-f7a74e1a524b.png">
Or it would be better to have it in one single file eg. `types.d.ts`? What would be your advice to keep types clean? 

I've notice the slots/named slots with passed props are not being recognised by Jet Brains, is it sth work in progress? Or shall I annotate the slots?

**Comment by qmhc** [comment]
Fixed and improved in `3.0.0`.
